src/redux/services/api.ts

```diff
@@ -58,11 +59,43 @@ export const api = createApi({
     // GET: Search for customers with search params
     getCustomers: builder.query<CustomerListItem[], string>({
       query: (params) => `customers?${params}`,
+      providesTags: [{ type: 'Customer', id: 'LIST' }],
     }),
 
     // GET: Fetch single customer's details
     getCustomerById: builder.query<Customer, string>({
       query: (id) => `customers/${id}`,
```

Comment:
```suggestion
      query: (id) => `customers/${id}/`,
```
Not related to the changes in this PR, but happened to notice this URL is slightly wrong.

Reply:
Fixed